pkg/antctl/raw/multicluster/upgrade/clusterset.go

```diff
@@ -0,0 +1,154 @@
+	if o.namespace == "" {
+		o.namespace = metav1.NamespaceSystem
+	}
+	if o.k8sClient == nil {
+		kubeconfig, err := raw.ResolveKubeconfig(cmd)
+		if err != nil {
+			return err
+		}
+		o.k8sClient, err = client.New(kubeconfig, client.Options{Scheme: multiclusterscheme.Scheme})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func NewUpgradeClusterSetCommand() *cobra.Command {
+	cmdUpgrade := &cobra.Command{
+		Use: "clusterset",
+		Aliases: []string{
+			"clusterset",
+		},
+		Short:   "Upgrade Multi-cluster ClusterSets",
+		Args:    cobra.MaximumNArgs(1),
+		Example: upgradeExamples,
+		RunE:    runUpgrade,
+	}
+	o := &upgradeOptions{}
+	optionsUpgrade = o
+	cmdUpgrade.Flags().StringVarP(&o.namespace, "namespace", "n", "", "Namespace of ClusterSets")
+
+	return cmdUpgrade
+}
+
+func runUpgrade(cmd *cobra.Command, args []string) error {
+	err := optionsUpgrade.validateAndComplete(cmd)
+	if err != nil {
+		return err
+	}
+
+	clusterSets := &mcv1alpha2.ClusterSetList{}
+	ctx := context.Background()
+	err = optionsUpgrade.k8sClient.List(ctx, clusterSets, &client.ListOptions{Namespace: optionsUpgrade.namespace})
+	if err != nil {
+		return err
+	}
+	clusterSetsSize := len(clusterSets.Items)
+	if clusterSetsSize == 0 {
+		fmt.Fprintf(cmd.OutOrStdout(), "No ClusterSet found in Namespace %s, skip upgrading\n", optionsUpgrade.namespace)
+		return nil
+	}
+
+	existingClusterSet := clusterSets.Items[0]
+
+	// Do nothing if the existing ClusterSet already contains ClusterID.
+	if existingClusterSet.Spec.ClusterID != "" {
+		fmt.Fprintf(cmd.OutOrStdout(), "The ClusterSet %s in Namespace %s is already a new version, skip upgrading\n", existingClusterSet.Name, existingClusterSet.Namespace)
+		return nil
+	}
+
+	clusterClaims := &mcv1alpha2.ClusterClaimList{}
+	err = optionsUpgrade.k8sClient.List(ctx, clusterClaims, &client.ListOptions{Namespace: optionsUpgrade.namespace})
+	if err != nil {
+		fmt.Fprintf(cmd.OutOrStdout(), "Error when getting ClusterClaims in Namespace %s, you can retry upgrade later\n", optionsUpgrade.namespace)
+		return err
+	}
+
+	var clusterID, clusterSetID string
+	if len(clusterClaims.Items) == 0 {
+		fmt.Fprintf(cmd.OutOrStdout(), "No ClusterClaim found in Namespace %s, skip upgrading\n", optionsUpgrade.namespace)
+		return nil
+	}
+	for _, cc := range clusterClaims.Items {
+		if cc.Name == mcv1alpha2.WellKnownClusterClaimID {
+			clusterID = cc.Value
+		}
+		if cc.Name == mcv1alpha2.WellKnownClusterClaimClusterSet {
+			clusterSetID = cc.Value
+		}
+	}
+
+	if clusterID == "" || clusterSetID == "" {
+		fmt.Fprintf(cmd.OutOrStdout(), "ClusterID or ClusterSet ID is missing in ClusterClaims in Namespace %s, skip upgrading\n", optionsUpgrade.namespace)
+		return nil
+	}
+
+	if clusterID != "" && clusterSetID != "" {
+		oldClusterSet := existingClusterSet
+		existingClusterSet.Spec.ClusterID = clusterID
+		// Always replace the existing ClusterSet with a new version of ClusterSet
```

Comment:
@jianjuns when I was trying to modify the existing ClusterSet via antctl, I found that it's not allowed to add a new field by kubeconfig user. There is a ClusterSet webhook to validate the user info in old version https://github.com/antrea-io/antrea/blob/release-1.12/multicluster/cmd/multicluster-controller/clusterset_webhook.go#L67-L76.
So we can only delete and recreate a new version of ClusterSet CR via antctl.
Let me know if you have any concern about this. I am still working on the unit test and doc update.